### ISSUE TITLE
Can you fill a water bottle with drinking water here?

### ISSUE THREAD

**Issue by mcliquid** [streetcomplete/StreetComplete#4110]
### General
Affected tag(s) to be modified/added: [bottle](https://wiki.openstreetmap.org/wiki/Key:bottle)
Question asked: **Can you fill a water bottle with drinking water here?**

### Checklist
Checklist for quest suggestions (see [guidelines](https://github.com/streetcomplete/StreetComplete/blob/master/QUEST_GUIDELINES.md)):
- [X] 🚧 **To be added tag is established and has a useful purpose:** There are currently 13,181 uses ([taginfo](https://taginfo.openstreetmap.org/keys/?key=bottle)) which is comparatively low. To my knowledge, this tag is not yet particularly used in software, but would have potential for it. Usage has been increasing fast since 2020.
- [X] 🤔 **Any answer the user can give must have an equivalent tagging (Quest should not reappear to other users when solved by one):** It's common to tag yes or no. The share of "No" is currently around 15%.
- [X] 🐿️ **Easily answerable by everyone from the outside but a survey is necessary:** The understanding to tag should be present, even though drinking bottles can be different sizes. Whether a bottle can be filled up there, one should already be able to recognize when passing, since fountains are largely outside.
- [X] 💤 **Not an overwhelming percentage of quests have the same answer (No spam):** From my experience, drinking water points are not found on every corner. There are countries like Italy where there are proportionally more, but even here it is not spammy. Also, I would say that the majority of drinking water points allow you to fill a bottle. So the "no" rate should be rather low.
- [X] 🕓 **Applies to a reasonable number of map data (Worth the effort):** As of today, there are 244,022 drinking water points without bottle=* value worldwide. 

### Ideas for implementation

**Element selection:** 

- [amenity=drinking_water](https://wiki.openstreetmap.org/wiki/Tag:amenity%3Ddrinking_water) AND [bottle!=*](https://wiki.openstreetmap.org/wiki/Key:bottle)
- [man_made=water_tap](https://wiki.openstreetmap.org/wiki/Tag:man_made=water_tap) AND [bottle!=*](https://wiki.openstreetmap.org/wiki/Key:bottle) AND [drinking_water=yes](https://wiki.openstreetmap.org/wiki/Key:drinking_water)
- [man_made=water_well](https://wiki.openstreetmap.org/wiki/Tag:man_made=water_well) AND [bottle!=*](https://wiki.openstreetmap.org/wiki/Key:bottle) AND [drinking_water=yes](https://wiki.openstreetmap.org/wiki/Key:drinking_water) 
**TBD**: To enable meaningful tagging, a strict selection of sub-tags should probably be chosen. "drinking_water=yes" is mandatory, but probably others like `pump_mechanism=piston` and `mechanical_driver=manual` is needed.
- [amenity=fountain](https://wiki.openstreetmap.org/wiki/Tag:amenity=fountain)  AND [bottle!=*](https://wiki.openstreetmap.org/wiki/Key:bottle) AND [drinking_water=yes](https://wiki.openstreetmap.org/wiki/Key:drinking_water): If a fountain has a "drinking_water=yes", it should be possible to fill a bottle there.
- [man_made=drinking_fountain](https://wiki.openstreetmap.org/wiki/Tag:man_made=drinking_fountain) AND [bottle!=*](https://wiki.openstreetmap.org/wiki/Key:bottle)
**TBD**: From my point of view, the tag implies that there is only a small jet of water shooting upwards at which it should be impossible to fill a bottle.
- [natural=spring](https://wiki.openstreetmap.org/wiki/Tag:natural=spring) AND [bottle!=*](https://wiki.openstreetmap.org/wiki/Key:bottle)
**TBD**: If the spring looks like the one in the wiki (a pipe coming out of the middle of the mountain and carrying water), then it makes sense. But I suspect very few springs look like that.

**Metadata needed:** 
I have made a OverPass query that requests all of the above cases in my county: https://overpass-turbo.eu/s/1jie
That would be a little more than 500.

**Proposed UI:**
I would choose a simple yes/no question here and choose a bottle=yes or bottle=no according to the answer.
And I would choose a generic bicycle bottle as icon like this one: https://www.svgrepo.com/svg/73486/bottle

**Impediments to discuss:**
- No drinking water: I would formulate the question as explicitly as possible so that it really emerges that you can fill drinking water into a bottle there. How would we then cover the case that there is no drinking water there at all? Ideally, we could trigger the drinking water quest when the user chooses "Not answerable".
- I have seen some fountains where it would be technically possible to fill a bottle, but the water only runs out in droplets, which would lead to the process taking forever until the bottle is full. Would a new tagging variant like bottle=limited be appropriate here?

**Comment by andrewharvey** [comment]
I would say we must include "easily" in the question, "Can you **easily** fill a water bottle with drinking water here?" given that's how it's documented on the wiki. Otherwise you'll have people answering technically yes, even though it's practically no.

**Comment by mnalis** [comment]
I guess the tag is not used much, because it is vaguely defined. I am drinking water freak myself, finding `drinking_water` tag one of the most important things to map on OSM, yet I rarely if ever map `bottle=*`. Problems are at least:

- it talks about "water bottle" without defining what exactly it means. Do we talk about (quite common here) 1.5L water bottles which you would use for longer trips (like in motor vehicle or in bicycle panniers)? Or of smaller 0.75 variety (the kind I use exclusively on bicycle, because it holds enough water and still fits nicely in standard quick-to-reach bottle holder). Or hiking types, which are usually 0.5L or 0.3L? It makes a huge difference, especially for tap/sink clearance). So same water source might **_easily fit_** one bottle, while at the same time **_impossible_** to fit another.
- it uses undefined and vague terms such as "easy". It is highly subjective.
- "easy" does not equal "quick". If I can just put the bottle on the ground and it is filled in **5 minutes**, it is quite "easy", but not particularly "quick". On the other hand, one might be able to fill the bottle in just **5 seconds**, but they need to stand on one leg extending themselves on 45 degrees, trying to hold on bare wet rock with one hand while filling the bottle with another - it is quite "quick", but not "easy" at all.
- wiki is quite terse and poorly defined overall

In short, I just mark `drinking_water` on most reasonable water sources (which is majority of them, under my definition of "reasonable" - i.e. less then 5 minute stop to fill water bottle), and leave a Note (and later fill the `description` or other tags) to describe more problematic ones in more detail (e.g. hard to reach, small clearance, seasonal only, very low flow rate, intended for hose connection so sprays wildly around, hard to use one-handed like manual pumps, vertical or angled jets harder to catch in bottles etc.)

**Comment by rhhsm** [comment]
I think it is reasonable to have this quest for public drinking water facilities in urban areas, which is what the amenity=drinking_water tag mostly has in mind, but not of natural sources of drinking water (natural=spring) in rural and mountainous areas where the answer is usually yes (so violating the No Spam criterion). Like mnalis, I'm a fan of adding these rural sources on the map, but I never added bottle filling info. I would welcome a quest to tag such sources as seasonal (intermittent=* )

**Comment by mnalis** [comment]
> I think it is reasonable to have this quest for public drinking water facilities in urban areas

how do you reconcile [4 issues brought up before](https://github.com/streetcomplete/StreetComplete/issues/4110#issuecomment-1155003001), though?

> 'm a fan of adding these rural sources on the map, but I never added bottle filling info. I would welcome a quest to tag such sources as seasonal (intermittent=* ).

Ummm, should this be a separate issue then, if we're not talking about `bottle=yes` but `intermittent=*` here?

> On urban sources, maybe it could be asked only for sources where bottle filling is doubtful, i.e. on man_made=drinking_fountain

Note that `man_made=drinking_fountain` is [hopeless](https://wiki.openstreetmap.org/wiki/Proposed_features/Deprecate_man_made%3Ddrinking_fountain). (`fountain=drinking` might be an working alternative, but, meh.)

**Comment by rhhsm** [comment]
> > I think it is reasonable to have this quest for public drinking water facilities in urban areas
> 
> how do you reconcile [4 issues brought up before](https://github.com/streetcomplete/StreetComplete/issues/4110#issuecomment-1155003001), though?

"Easily fill" is not an objective criterion indeed, but I don't think this is a problem. It will depend on local conditions, such as which kind of water bottle people commonly want to fill (probably the 0.5L PET mineral water bottle?). I think the large majority of drinking water sources do not have the problems you mention in your third bullet.

> > 'm a fan of adding these rural sources on the map, but I never added bottle filling info. I would welcome a quest to tag such sources as seasonal (intermittent=* ).
> 
> Ummm, should this be a separate issue then, if we're not talking about `bottle=yes` but `intermittent=*` here?

On second thoughts, this would not be very suitable for an SC quest because it is answerable only in a particular season (end of summer for instance for natural springs). It is very useful to know a bit more about the reliability of drinking water sources when on a multi-day hike in the mountains.

> > On urban sources, maybe it could be asked only for sources where bottle filling is doubtful, i.e. on man_made=drinking_fountain
> 
> Note that `man_made=drinking_fountain` is [hopeless](https://wiki.openstreetmap.org/wiki/Proposed_features/Deprecate_man_made%3Ddrinking_fountain). (`fountain=drinking` might be an working alternative, but, meh.)

Maybe only on fountain=bubbler then, as it is the one most likely to be problematic for bottle filling. All others are probably mostly bottle=yes so it would be quite spammy.
